source/common/grpc/google_async_client_impl.h

```diff
@@ -184,6 +184,8 @@ class GoogleAsyncClientImpl final : public RawAsyncClient, Logger::Loggable<Logg
   RawAsyncStream* startRaw(absl::string_view service_full_name, absl::string_view method_name,
                            RawAsyncStreamCallbacks& callbacks,
                            const Http::AsyncClient::StreamOptions& options) override;
+  // This does not use the worker thread's dispatcher, so we must return nullptr.
+  Event::Dispatcher* dispatcher() override { return nullptr; }
```

Comment:
I don't think you need to plumb the dispatcher from the gRPC client. The filter should be able to obtain this from stream decoder callbacks, see https://github.com/envoyproxy/envoy/blob/419428a82945a1a2ab0ab48e054e76ee851c5450/include/envoy/http/filter.h#L174

Reply:
1. It's fine to add to the `check` args a dispatcher ref. It could also be done at client creation time, but I think we will make this TLS in the near future, so fine to add to `check`. 
2. Not sure I fully grok how the Google client's native timeout interacts with the choice of how we plumb dispatcher.

Reply:
1. sounds good
2. The auth-client doesn't know if it has a google grpc client or native grpc client. if it gets a dispatcher in the check method args, it will always use it for timeout purposes (as it can't tell the grpc clients apart). If an optional dispatcher is returned from the grpc client, the auth-client will only use it if it exists (i.e. will not use it for google grpc, instead it will set the request options timeout).

Reply:
I see on (2). Two options I see:
A. Avoid using Google gRPC client timeout, just do a single local timeout in ext_authz no matter what.
B. Fix Envoy gRPC client to do a timeout consistent with Google gRPC. I'm fine punting this to a TODO for the future. It seems this would reduce confusion systematically though.

Reply:
sounds good! thank you for the feedback